### PR TITLE
Minor fixes

### LINKED ISSUE #77: Saved Response JSON contains \uxxxx encoded symbols if using OhsomeResponse.to_json() function
Example (geometry extraction response part):
```
      "properties": {
        "@osmId": "way/394983845",
        "@snapshotTimestamp": "2021-09-01T00:00:00Z",
        "demolished:highway": "unclassified",
        "description": "ehemalige R\u00f6merstra\u00dfe nach Ladenburg, Verlauf teilweise noch erkennbar",
        "historic": "yes"
      }
```
Original Response JSON:
```
    "properties" : {
      "@osmId" : "way/394983845",
      "@snapshotTimestamp" : "2021-09-01T00:00:00Z",
      "demolished:highway" : "unclassified",
      "description" : "ehemalige Römerstraße nach Ladenburg, Verlauf teilweise noch erkennbar",
      "historic" : "yes"
    }
```

Here is the line of code causing this:
https://github.com/GIScience/ohsome-py/blob/b405c74f6dd47c347a457bc0baf57424845e7ebb/ohsome/response.py#L118

To avoid this conversion of the original ohsome-api response you could use `ensure_ascii=False` flag:
```python
 with open(outfile, "w", encoding="utf-8") as dst:
            json.dump(self.data, dst, indent=2, ensure_ascii=False)
```

### LINKED ISSUE #78: json decode error on wrong api url
using `https://api.ohsome.org/elemets/geometry` instead of `https://api.ohsome.org/v1/elements/geometry` as api url wil cause a json-decode error because the response is an html error message.

can we catch that somehow?

### LINKED ISSUE #80: empty result will throw TypeError for .as_dataframe()
Requesting a geodataframe will end at https://github.com/GIScience/ohsome-py/blob/317059d3ac15151e954b6565eaa87366d158e82c/ohsome/response.py#L107 if no data is returned (empty geodataframe)

### LINKED ISSUE #82: documentation for user_agent missing
user_agent parameter not described in documentation: https://github.com/GIScience/ohsome-py/blob/317059d3ac15151e954b6565eaa87366d158e82c/ohsome/clients.py#L33-L41

### LINKED ISSUE #83: TypeError: __init__() got an unexpected keyword argument 'allowed_methods'
Calling 

```
client.elements.geometry.post(bpolys=parks_dd, filter="(amenity=bench) and (type:node)")
```
throws the following error 

```
TypeError: __init__() got an unexpected keyword argument 'allowed_methods'
```

Apperently urllib3 needs to be >=1.26. See [this issue](https://github.com/plamere/spotipy/issues/647).

### LINKED ISSUE #85: Update authors in pyproject.toml
The authors should be updated to the contributors of this project. Alternatively only specifying the ohsome team or giscience team may be an option: https://github.com/GIScience/ohsome-quality-analyst/blob/85bd7644ea9608ad206c4b2c6691d231d33b3249/workers/pyproject.toml

### PR DESCRIPTION
**Pull request by redfrexx**
Includes some minor bug fixes: 

- Handle responses with no features without geopandas warning (closes #80) 
- added documentation for user_agent (closes #82) 
- Fixed encoding when writing response to json (closes #77) 
- Handles exception when invalid ohsome API URL is given (closes #78) 
- updated authors in pyproject.toml  (closes #85)
- added urllib3 as dependency (closes #83 #91)

### CODE REVIEW

**Comment by SlowMo24** [comment]
I will have a look at the changes but I guess you want to fix the failing checks first?

File: ohsome/test/test_exceptions.py [hunk 1]
```diff
@@ -55,6 +55,23 @@ def test_invalid_url():
...
+    """
+    base_api_url = "https://api.ohsome.org/"
+    bboxes = "8.7137,49.4096,8.717,49.4119"
+    time = "2018-01-01"
+    fltr = "name=Krautturm and type:way"
+
+    client = ohsome.OhsomeClient(base_api_url=base_api_url, log=False)
+    with pytest.raises(ohsome.OhsomeException) as e_info:
+        client.elements.count.post(bboxes=bboxes, time=time, filter=fltr)
+    assert "Invalid response" in e_info.value.message
```
Comment (SlowMo24) [inline ohsome/test/test_exceptions.py:72]: I think you rather want to to assert that the cause is reported/guessed (invalid url)?

File: ohsome/test/test_response.py [hunk 2]
```diff
@@ -323,3 +324,22 @@ def test_contributions_latest(custom_client):
...
+    time = "2015-01-01,2016-01-01"
+    filter = "name=Krautturm1 and type:way"
+
+    client = custom_client
+    response = client.elements.bbox.post(bboxes=bboxes, time=time, filter=filter)
+    with pytest.warns(None) as record:
+        result = response.as_dataframe()
+
+    assert isinstance(result, gpd.GeoDataFrame)
+    assert len(result) == 0
```
Comment (SlowMo24) [inline ohsome/test/test_response.py:344]: `result.empty` is the suggested method by pandas, I guess

**Review by SlowMo24 — APPROVED** [review]
looks good, nice test driven development!

File: ohsome/test/test_exceptions.py (hunk 1 above)
Comment (redfrexx) [reply ohsome/test/test_exceptions.py]: ah yes, makes more sense. I've changed it. Thanks!

File: ohsome/test/test_response.py (hunk 2 above)
Comment (redfrexx) [reply ohsome/test/test_response.py]: yes looks nicer. :) Thanks!

**Pull request merged**